torch/csrc/jit/tensorexpr/scripts/bisect.py

```diff
@@ -0,0 +1,68 @@
+import subprocess
+import click
+
+
+def test(cmd, limit):
+    print(f"Testing PYTORCH_JIT_OPT_LIMIT=tensorexpr_fuser={limit} {cmd}")
+    p = subprocess.run(
+        f"PYTORCH_JIT_OPT_LIMIT=tensorexpr_fuser={limit} {cmd}",
+        shell=True,
+        capture_output=True,
+        encoding="utf-8",
+    )
+    print(p.stdout)
+    f = "INTERNAL ASSERT FAILED"
+    if f in p.stdout or f in p.stderr:
+        print("skip")
+        return -1
+    if p.returncode == 0:
+        print("good")
+        return 1
+    print("bad")
+    return 0
+
+
+@click.command()
+@click.option("--cmd")
+def bisect(cmd):
+    last_good = 0
+    first_bad = 10000
+    skips = set()
+
+    # Test if there are any unskipped commits in (last_good, first_bad)
+    def keep_going():
+        for limit in range(last_good + 1, first_bad):
+            if limit not in skips:
+                return True
+        return False
+
+    while keep_going():
+        test_limit = test_mid = (last_good + first_bad) // 2
+        val = -1
+
+        # Scan forward from mid towards bad.
+        while test_limit <= first_bad and val == -1:
+            val = test(cmd, test_limit)
```

Comment:
There could be duplicated runs for a `test_limit` that has already been added in `skips`. The following check can avoid the duplication:
```
if test_limit in skips:
  test_limit = test_limit + 1
```

Reply:
Or if I understand correctly, whenever we hit a `test_limit` in `skips`, then all test_limits in `range(test_limit, first_bad)` are in `skips` too. So break the while loop in this case?
```
if test_limit in skips:
  break
```

Reply:
I'm not entirely sure about this case :).  skipping the re-test is probably a legit optimization although I'm not sure whether that case ends up being hit ever b/c of the way the search space narrows each iteration of binary search...

Reply:
In theory the case can happen. But maybe in reality it's rare to happen and can be omitted.

An example for one `test_limit` being hit twice in theory: assuming last_good, first_bad = 0, 100
  1st iteration of `while keep_going()`:
  ```
  mid = 50, all test_limits in (mid, 80) returns -1, and test_limit=80 returns 0 so last_good, first_bad = 0, 80
  ```
  2nd iteration of `while keep_going()`:
  ```
  mid = 40, all test_limits in (mid, 80) returns -1 so it starts scanning back towards good ...
  ```
In the 2nd iteration, test_limits in (50, 80) are tested the second time.